assets/chat/js/menus.js

```diff
@@ -452,10 +453,289 @@ class ChatWhisperUsers extends ChatMenu {
+                            reason : reasonValue
+                        };
+                        if(/^perm/i.test(durationValue))
+                            payload.ispermanent = true
+                        else
+                            payload.duration = this.chat.parseTimeInterval(durationValue)
+            
+                        payload.banip = banType === 'IPBAN'
+            
+                        this.chat.source.send('BAN', payload)
+                    } else {
+                        MessageBuilder.error('Providing a reason is mandatory').into(this.chat)
+                    }
+                    break;
+                case 'mute':
+                    const duration = (durationValue) ? this.chat.parseTimeInterval(durationValue) : null
+                    if (duration && duration > 0) {
+                        this.chat.source.send('MUTE', {data: this.clickedNick, duration: duration})
+                    } else {
+                        this.chat.source.send('MUTE', {data: this.clickedNick})
+                    }
+                    break
+                default:
+                    break
+            }
+            super.hide()
+        }
+    }
+
+    addContent(message){
+        this.messageArray = [message]
+
+        const prettyNick = message.find('.user')[0].text
+        const nick = message.data('username')
+        const usernameFeatures = message.find('.user')[0].attributes.class.value
+
+        const featuresList = this.buildFeatures(nick, usernameFeatures)
+        if (featuresList === '') {
+            this.flairList.hide()
+            this.flairSubheader.style.display = 'none'
+        } else {
+            this.flairList.show()
+            this.flairSubheader.style.display = ''
+        }
+
+        const messageList = this.createMessages()
+        if (messageList.length === 1) {
+            this.messagesSubheader.innerText = 'Selected message:'
+        } else {
+            this.messagesSubheader.innerText = 'Selected messages:'
+        }
+
+        this.header.text("")
+        this.header.attr('class', 'username')
+        this.messagesContainer.empty()
+        this.flairList.empty()
+
+        this.header.text(prettyNick)
+        this.header.addClass(usernameFeatures)
+        this.flairList.append(featuresList)
+        messageList.forEach(element => {
+            this.messagesContainer.append(element)
+        })
+
+        super.redraw()
+    }
+
+    buildFeatures(nick, messageFeatures){
+        const user = this.chat.users.get(nick)
+        const messageFeaturesArray = messageFeatures.split(' ').filter(e => e !== 'user' && e !== 'subscriber')
+        const features = (user !== undefined) ? this.buildFeatureHTML((user.features.filter(e => e !== 'subscriber') || [])) : this.buildFeatureHTML(messageFeaturesArray)
+        return features !== '' ? `<span class="features">${features}</span>` : ''
+    }
+
+    hasModPowers(user) {
```

Comment:
I think the `ChatUser` class is a better fit for this method.

Reply:
Good call! Moved it :+1: